utils/driver_options/driveroptions.cpp

```diff
@@ -6,20 +6,110 @@
 #include <string>
 #include <iostream>
 #include <sstream>
+#include <vector>
+#include <map>
 #include "utils/options/options.hpp"
 #include "utils/driver_options/driveroptions.hpp"
 
-
 /**
  * @brief      Extracts the relevant driver options from the config into class variables.
  */
 DriverOptions::DriverOptions() {
     Options::parse_config_file();
 
-    devices_ = config_pt.get<std::string>("devices");
+    devices_ = config_pt.get<std::string>("device_options");
+    
+    auto n200_list = config_pt.get_child("n200s");
+    // These maps are sorted by their keys (device number / int antenna number)
+    std::map<uint32_t, std::string> devices_map;    // Maps device number to IP address
+    std::map<uint32_t, bool> rx_map;                // Maps device number to rx flag
+    std::map<uint32_t, bool> tx_map;                // Maps device number to tx flag
+    std::map<uint32_t, uint32_t> int_antenna_map;   // Maps interferometer antenna number to device number
+
+    // Count up antennas while looping through N200 array
+    main_antenna_count_ = 0;
+    interferometer_antenna_count_ = 0;
+
+    // Iterate through all N200s in the json array
+    for (auto n200 = n200_list.begin(); n200 != n200_list.end(); n200++)
+    {
+        // Start iterator on first item (addr)
+        auto iter = n200->second.begin();
+
+        // Get n200 IP address
+        auto addr = iter->second.data();
+
+        // Get rx, tx, and rx_int flags
+        iter++;
+        bool rx = (iter->second.data().compare("true") == 0);
+        iter++;
+        bool tx = (iter->second.data().compare("true") == 0);
+        iter++;
+        bool rx_int = (iter->second.data().compare("true") == 0);
+
+        // If current n200 is transmitting, receiving, or receiving from interferometer, add to devices
+        if (tx || rx || rx_int)
+        {
+            // Get device number. Devices are sorted by the main antenna they are connected to
+            iter++; 
+            auto device_num = boost::lexical_cast<uint32_t>(iter->second.data());
+
+            // Add the address, tx flag, and rx flag to the respective dictionaries keyed with the device num
+            devices_map[device_num] = addr;
+            tx_map[device_num] = tx;
+            rx_map[device_num] = rx;
+
+            // If N200 has interferometer, map device number to interferometer antenna number
+            if (rx_int) {
+                iter++;
+                auto int_antenna_num = boost::lexical_cast<uint32_t>(iter->second.data());
+                int_antenna_map[int_antenna_num] = device_num;
+                interferometer_antenna_count_++;
+            }
+            if (tx || rx) {
```

Comment:
I think this if statement (tx || rx) can be moved out of the outer if statement scope